lib/workload/stateless/stacks/icav2-event-translator/deploy/stack.ts

```diff
@@ -0,0 +1,100 @@
+  eventBusName: string;
+  /** vpc ann SG for lambda function */
+  vpcProps: VpcLookupOptions;
+  lambdaSecurityGroupName: string;
+}
+
+export class Icav2EventTranslatorStack extends Stack {
+  private readonly vpc: IVpc;
+  private readonly lambdaSG: ISecurityGroup;
+  private readonly mainBus: IEventBus;
+  private readonly dynamoDBTable: ITableV2;
+  private readonly lambdaRuntimePythonVersion = Runtime.PYTHON_3_12;
+
+  constructor(scope: Construct, id: string, props: StackProps & Icav2EventTranslatorStackProps) {
+    super(scope, id, props);
+    this.mainBus = EventBus.fromEventBusName(this, 'EventBus', props.eventBusName);
+    this.vpc = Vpc.fromLookup(this, 'MainVpc', props.vpcProps);
+    this.dynamoDBTable = TableV2.fromTableName(
+      this,
+      'Icav2EventTranslatorDynamoDBTable',
+      props.icav2EventTranslatorDynamodbTableName
+    );
+    this.lambdaSG = SecurityGroup.fromLookupByName(
+      this,
+      'LambdaSecurityGroup',
+      props.lambdaSecurityGroupName,
+      this.vpc
+    );
+
+    this.createICAv2EventTranslator(props);
+  }
+
+  private createICAv2EventTranslator(props: Icav2EventTranslatorStackProps) {
+    const EventTranslatorFunction = new PythonFunction(this, 'EventTranslator', {
+      entry: path.join(__dirname, '../translator_service'),
+      runtime: this.lambdaRuntimePythonVersion,
+      environment: {
+        TABLE_NAME: props.icav2EventTranslatorDynamodbTableName,
+        EVENT_BUS_NAME: props.eventBusName,
+      },
+      vpc: this.vpc,
+      vpcSubnets: { subnets: this.vpc.privateSubnets },
+      securityGroups: [this.lambdaSG],
+      architecture: Architecture.ARM_64,
+      timeout: Duration.seconds(28),
+      index: 'icav2_event_translator.py',
+      handler: 'handler',
+    });
+
+    this.dynamoDBTable.grantReadWriteData(EventTranslatorFunction);
+
+    // Create a rule to trigger the Lambda function from the EventBus ICAV2_EXTERNAL_EVENT
+    const rule = new Rule(this, 'Rule', {
+      eventBus: this.mainBus,
+      eventPattern: {
+        account: [Stack.of(this).account],
+        detail: {
+          'ica-event': {
+            eventCode: ['ICA_EXEC_028'],
+            projectId: [{ exists: true }],
+            payload: {
+              pipeline: {
+                code: [{ prefix: 'BclConvert' }],
+              },
+            },
+          },
+        },
+      },
+    });
+
+    rule.addTarget(
+      new LambdaFunction(EventTranslatorFunction, {
+        maxEventAge: Duration.seconds(60), // Maximum age for an event to be retried, Member must have value greater than or equal to 60 (Service: EventBridge)
+        retryAttempts: 3, // Retry up to 3 times
+      })
+    );
+
+    // Optional: If the Lambda function needs more permissions for the DynamoDB table and the event bus
+    EventTranslatorFunction.addToRolePolicy(
+      new PolicyStatement({
+        actions: ['events:PutEvents', 'dynamodb:PutItem', 'dynamodb:GetItem', 'dynamodb:Scan'],
```

Comment:
I suppose the `grantReadWriteData` function defined in row 64 cover this already?

Reply:
replace these redundancies with new Query permission.

Reply:
Ahh, what I meant the `grantReadWriteData` that you defined in line 64 should cover relevant read action for that lambda to that table. So I don't think you need this `Query` permission at all.

Ref: https://github.com/aws/aws-cdk/blob/205163fc0d2cac84d3d746a98c393e137f0e2388/packages/aws-cdk-lib/aws-dynamodb/lib/table-v2-base.ts#L170-L185

Reply:
just saw this. sure, definitely, will remove these redundant stuff.